From 590fdfc0339a2d82f30927b7c4e35b5e24d2dcc8 Mon Sep 17 00:00:00 2001
From: Roger Zanoni <rzanoni@igalia.com>
Date: Mon, 30 Oct 2023 08:14:34 -0300
Subject: [PATCH 24/33] [M118-fix] Fix comparison in HostResolverCache

Fixes the string_view/basic_strinc comparison:

error: invalid operands to binary expression
('tuple<...basic_string_view<char> &, ...>' and 'tuple<const std::basic_string<char> &, ...')
      return std::tie(lhs.domain_name, *lhs.network_anonymization_key) <
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^

Upstream-Status: Inappropriate
Signed-off-by: Roger Zanoni <rzanoni@igalia.com>
---
 net/dns/host_resolver_cache.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/dns/host_resolver_cache.h b/net/dns/host_resolver_cache.h
index 934e0682c9864..2bb147de0ff19 100644
--- a/net/dns/host_resolver_cache.h
+++ b/net/dns/host_resolver_cache.h
@@ -120,7 +120,7 @@ class NET_EXPORT HostResolverCache final {
   struct Key {
     ~Key();
 
-    std::string domain_name;
+    base::StringPiece domain_name;
     NetworkAnonymizationKey network_anonymization_key;
   };
 
-- 
2.42.1

